test/Activator.test.js

```diff
@@ -0,0 +1,92 @@
+var assert = require('assert');
+var sinon = require('sinon');
+var jsdom_global = require('jsdom-global');
+
+var canvasMockify = require('./canvas-mock');
+var Activator = require('../lib/shared/Activator');
+
+
+describe('Activator', function () {
+  beforeEach(function() {
+    this.jsdom_global = jsdom_global(
+      "<div id='mynetwork'></div>",
+      { skipWindowCheck: true}
+    );
+    canvasMockify(window);
+    this.container = document.getElementById('mynetwork');
+  });
+
+  afterEach(function() {
+    this.jsdom_global();
+    this.container.remove();
+    this.container = undefined;
+  });
+
+  describe('constructor', function () {
+
+    it('sets defaults', function () {
+      var activator = new Activator(this.container);
+      assert.equal(activator.active, false);
+    });
+
+    it('creates overlay', function () {
+      var activator = new Activator(this.container);
+      assert.equal(activator.dom.container.children[0].className, 'vis-overlay');
+    });
+  });
+
+  describe('activate', function () {
+    it('emits an `activate` event', function () {
+      var eventSpy = sinon.spy();
+      var activator = new Activator(this.container);
+      activator.on('activate', eventSpy);
+      activator.activate();
+      assert.equal(activator.active, true);
+      assert(eventSpy.called, 'Event did not fire.');
+      assert(eventSpy.calledOnce, 'Event fired more than once');
```

Comment:
In this case `sinon` provides a spy, which acts as a sort of accumulator of meta data, about how it was used. 
Similar to a mock, except that I'm able to verify whether or not it was called, how many times, and what arguments it was called with.
This test ensures that if code changes were introduced that prevented the callback from being triggered, or caused it to be triggered multiple times, the test would fail.